tools/ports/__init__.py

```diff
@@ -393,10 +393,20 @@ def add_deps(node):
     add_deps(port)
 
 
-def handle_use_port_arg(settings, name):
+def handle_use_port_arg(settings, arg):
+  args = arg.split('?', 1)
+  name, options = args[0], args[1] if len(args) == 2 else None
   if name not in ports_by_name:
     utils.exit_with_error(f'Invalid port name: {name} used with --use-port')
   ports_needed.add(name)
+  if options:
+    port = ports_by_name[name]
+    if not hasattr(port, 'handle_options'):
+      utils.exit_with_error(f'Invalid options for port {name}: No options available')
+    else:
+      error = port.handle_options(options)
+      if error is not None:
```

Comment:
How about we just have the port itself call `exit_with_error` if it doesn't like the options, and avoid return anything at all there.

Reply:
I agree there is argument to be made there.  We tend to err on the side of "bail hard and bail early whenever possible" which mean you need to write so much error checking code and keeps the calling code straight forward.

Almost 100% of emscripten testing (like most compilers) is black box testing (we test the compiler by running it), so we are less concerned with internal unit testing.

Reply:
Sounds good. I used `exit_with_error`